src/internal/bytealg/bytealg.go

```diff
@@ -251,3 +251,13 @@ func IndexRabinKarp(s, substr string) int {
 	}
 	return -1
 }
+
+// MakeNoZero makes a slice of length and capacity n without zeroing the bytes.
+// It is the caller's responsibility to ensure uninitialized bytes
+// do not leak to the end user.
+func MakeNoZero(n int) []byte {
+	// Note: this does zero the buffer even though that's not necessary.
+	// For performance reasons we might want to change this (similar to the
+	// malloc function implemented in the runtime).
```

Comment:
Does anyone every do that? I don't.
I typically use these kinds of comments as a hint to future me (or future someone else). With a TODO, I know something is really missing that should be implemented. In this case I didn't use a TODO because it's working as intended, just a bit slower than possible (and therefore if this shows up in a profiler there's a clear hint how to improve things).